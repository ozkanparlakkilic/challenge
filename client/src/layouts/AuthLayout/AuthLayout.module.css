.auth_layout {
    background-color: #e5e2e2;
    border-radius: 20px;
    width: 400px;
    height: 600px;
    padding: 30px;
    display: flex;
    flex-direction: column;
}