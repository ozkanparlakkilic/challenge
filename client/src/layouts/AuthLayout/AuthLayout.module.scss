@import '@/styles/scss/_variables';
@import '@/styles/scss/_mixins';

.auth_layout {
    background-color: $auth-layout-color;
    border-radius: 20px;
    width: 400px;
    height: 600px;
    padding: 30px;
    @include flex-column;
}