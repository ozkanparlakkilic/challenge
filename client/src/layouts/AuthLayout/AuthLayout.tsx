import React from 'react'
import styles from './AuthLayout.module.scss'

interface AuthLayoutProps {
    children: React.ReactNode;
    classname?: string | '';
}

const AuthLayout = ({children,classname}:AuthLayoutProps) => {
  return (
    <div className={`${classname ?? ''} ${styles.auth_layout}`}>{children}</div>
  )
}

export default React.memo(AuthLayout)