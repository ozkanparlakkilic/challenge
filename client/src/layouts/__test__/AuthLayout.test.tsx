import React from 'react';
import { render } from '@testing-library/react';
import AuthLayout from '../AuthLayout/AuthLayout';

describe('Renderings', () => {
    it('should render the child component', () => {
      const childElement = ("<div>Testing</div>")
      const { container } = render(<AuthLayout children={childElement} />)
      expect(container).toMatchSnapshot();
    });
});