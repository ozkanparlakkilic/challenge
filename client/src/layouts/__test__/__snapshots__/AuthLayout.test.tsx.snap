// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderings should render the child component 1`] = `
<div>
  <div
    class=" auth_layout"
  >
    &lt;div&gt;Testing&lt;/div&gt;
  </div>
</div>
`;
